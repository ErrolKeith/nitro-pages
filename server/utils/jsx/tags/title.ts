import { makeJsx } from "./makeJsx";

export interface TitleProps {
  title: string;
  className?: string;
}

export function makeTitle(config: TitleProps) {
  return makeJsx("title", { className: config.className }, config.title);
}
