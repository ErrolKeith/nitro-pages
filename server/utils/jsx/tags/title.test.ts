import { it, expect } from "vitest";
import { makeTitle, TitleProps } from "./title";

it("can make a title tag", () => {
  const titleTestProps: TitleProps = { title: "test title" };
  const titleTest = makeTitle(titleTestProps);
  expect(titleTest).toMatchSnapshot();
});
