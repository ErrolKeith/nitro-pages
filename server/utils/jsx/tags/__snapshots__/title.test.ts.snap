// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a title tag 1`] = `
<title>
  test title
</title>
`;
